rust/src/jsonbuilder.rs

```diff
@@ -186,6 +186,15 @@ impl JsonBuilder {
         }
     }
 
+    // Unclose an object which must not be empty.
+    pub fn unclose(&mut self) -> Result<&mut Self, JsonError> {
+        match self.buf.pop() {
+            Some('}') => {}
+            _ => {return Err(JsonError::InvalidState);}
+        }
+        self.push_state(State::ObjectNth)?;
+        Ok(self)
+    }
```

Comment:
Its quite confusing to me why this is even required?  Wouldn't refactoring the calling code a little get rid of this need?

Reply:
answering in the next comment as they are both about the same thing...